pontoon/base/static/js/translate.js

```diff
@@ -973,10 +973,12 @@ var Pontoon = (function (my) {
       // Set default input values and limits
       var from = counts[0][0],
           to = counts[counts.length - 1][0];
+      if(!$('#from').data('id')) $('#from').val(Highcharts.dateFormat('%d/%m/%Y %H:%M', from));
+      if(!$('#to').data('id')) $('#to').val(Highcharts.dateFormat('%d/%m/%Y %H:%M', to));
 
-      $('#from').val(Highcharts.dateFormat('%d/%m/%Y %H:%M', from));
-      $('#to').val(Highcharts.dateFormat('%d/%m/%Y %H:%M', to));
-
+      if($('#from').data('id')||$('#to').data('id')){
+        return;
+      }
```

Comment:
I think not-reseting the time range inputs can be greatly simplified. You only need to wrap the code to set the `#from` and `#to` input fields into an `if` block. Something like this:

```js
      // Set default input values and limits if not set yet
      if (!$('#filter li.time-range').is('.selected')) {
        var from = counts[0][0],
            to = counts[counts.length - 1][0];

        $('#from').val(Highcharts.dateFormat('%d/%m/%Y %H:%M', from));
        $('#to').val(Highcharts.dateFormat('%d/%m/%Y %H:%M', to));
      }
```

And then you can get rid of all the traces of `data('id')` (also in the template file).